src/org/thoughtcrime/securesms/ReactionsHandler.java

```diff
@@ -0,0 +1,151 @@
+package org.thoughtcrime.securesms;
+
+
+import android.content.Context;
+import android.database.Cursor;
+import android.os.Parcel;
+
+import org.thoughtcrime.securesms.database.Address;
+import org.thoughtcrime.securesms.database.DatabaseFactory;
+import org.thoughtcrime.securesms.database.MessageReactionDatabase;
+import org.thoughtcrime.securesms.database.model.MessageRecord;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ReactionsHandler {
+    private MessageReactionDatabase reactionsDb;
+
+    public ReactionsHandler(Context context) {
+        this.reactionsDb = DatabaseFactory.getMessageReactionDatabase(context);
+    }
+
+    /**
+     * NOT TEST
+     * This method only used by the conversation fragment since it will automatically
+     * compute the reactorID's to the user itself, Please uses another method if the reaction made
+     * by the recipient. such method can be accessed directly from the MessageReactionDB class
+     * The reaction time will be set to the current system time in milliseconds
+     * @param record
+     * @param reaction
+     */
+    public void reactToMessage(MessageRecord record, String reaction, Long time) {
+        this.reactionsDb.reactToMessage(record, reaction, time);
+    }
+
+    /**
+     * NOT TESTED
+     * This method is ONLY used by JobReceivers since when a reaction received, there
+     * is no active MessageRecord Message object, all what is required the MessageHash being reacted,
+     * reaction, the reactionDate.
+     * This Method can also be used by Fragments however, the developer should handle the time
+     * of the reaction manually
+     * @param messageHash
+     * @param reaction
+     * @param reactionTime
+     * @param reactorID
+     * @param threadId
+     */
+    public void reactToMessage(String messageHash, String reaction, Long reactionTime, Address reactorID, Long threadId) {
+        this.reactionsDb.reactToMessage(messageHash, reaction, reactionTime, reactorID, threadId);
+    }
+
+    public boolean deleteReaction(MessageRecord record) {
+        return this.reactionsDb.removeReaction(record) > 0 ? true:false;
```

Comment:
`>` result is a boolean, the extra ternary is unnecessary

Reply:
```java
(this.reactionsDb.removeReaction(record) > 0) ? true:false;
```
is equivalent to

```java
(this.reactionsDb.removeReaction(record) > 0);
```

because 

```java
(true) ? true:false;
```

is redundant

Reply:
@g-harel why (true) ? true:false;  if a record has no reactions, this remove method will return false, I can change the signature to void and ignore the bool completely but it helps us know for debugging or production that something was actually removed from  the database